File: Logitude/AngularModules/AngularModules/Common/DataContracts/ManifestSL.ts
import { PartnerSL } from './PartnerSL';
import { PortSL } from './PortSL';
import { HouseSL } from './HouseSL';
import { ShipmentPickUpDeliverySL } from './ShipmentPickUpDeliverySL';
import {ShipmentPackagePM} from '../../Shipment/EntityPMs/ShipmentPackagePM';
export class ManifestSL {
    public ShipmentNumber: string;
    public MasterNumber: string;
    public DirectionId: string;
    public TransportModeId: string;

    public IncotermCode: string;
    public IncotermName: string;
    public IncotermId: string;
    public IncotermAddedManually: boolean;
    public TrailerNumber: string;
    MainCarriageATA: Date;
    Transshipment1ATA: Date;
    Transshipment2ATA: Date;
    Transshipment3ATA: Date;
    MAWBOBLDate: Date;
    public TruckNumber: string;
    public CarrierCode: string;
    public CarrierName: string;
    public CarrierId: string;
    public CarrierAddedManually: boolean;

    public FreightPrepaidCollectId: string;
    public OtherPrepaidCollectId: string;
    public ShipmentLevelCode: string;
    public Shipper: PartnerSL;
    public Consignee: PartnerSL;
    public Notify1: PartnerSL;
    public Notify2: PartnerSL;

    public ValueOfGoods: number;
    public GeneralDescriptionOfGoods: string;
    public HouseNumber: string;


    MasterDate: Date;
    public MainCarriageFromPort: PortSL;
    public MainCarriageToPort: PortSL;

    public Transshipment1FromPort: PortSL;
    public Transshipment1ToPort: PortSL;

    public Transshipment2FromPort: PortSL;
    public Transshipment2ToPort: PortSL;

    public Transshipment3FromPort: PortSL;
    public Transshipment3ToPort: PortSL;

    public EntityId: string;
    public LongMaster: string;


    public GrossWeight: number;
    public ChargeableWeight: number;
    public TEU: number;
    public PackagesQuantity: number;



    public Volume : number;
    public VolumetricWeight : number;
    public NumberOfPackages : number;
    public NumberOfContainers : number;
    public IsDangerous: boolean;
    
    public GrossWeightUnitCode: string;
    public ChargeableWeightUnitCode: string;
    public VolumeUnitCode: string;
    public DimensionsUnitCode: string;
    public ChargeableWeightInKG: number;
    public GrossWeightEdited: boolean;
    public GrossWeightInKG: number;

    public OrderGrossWeight: number;
    public ShipmentTypeId: string;
    public ShipmentTypeName: string;
    public AgentReference1: string;
    public AgentReference2: string;
    public AgentName: string;
    public SharedManifestRef: string;

    public ShipperName: string;
    public ConsigneeReference1: string;
    public ConsigneeReference2: string;
    public ShipperReference1: string;
    public ShipperReference2: string;
    public MainHarmonize: string;



    public MainCarriageAirlinePrefix: string;
    public MainCarriageVesselName: string;
    public MainCarriageVesselId: string;
    public MainCarriageVesselCode: string;
    public  MainCarriageVesselAddedManually :boolean;
    
    public HAWBDate: Date;
    public MainCarriageETD: Date;

    public  MainCarriageATD :Date;
    public MainCarriageETA: Date;




    public MainCarriageMAWBOBL: string;
    public MainCarriageMAWBOBLDate: string;
    public InterlineId: string;
    public InterlineCode: string;
    public InterlineName: string;
    public InterlineAddedManually: boolean;
    public MainCarriageCarrierNumber: string;
 


    public Transshipment1CarrierName: string;
    public Transshipment1CarrierCode: string;
    public Transshipment1CarrierId: string;
    public Transshipment1CarrierAddedManually: boolean;
    public Transshipment1AirlinePrefix: string;
    public Transshipment1ETD: Date;
    public Transshipment1ETA: Date;
    public Transshipment1ATD: Date;
    public Transshipment1MAWBOBL: string;
    public Transshipment1CarrierNumber: string;
    public Transshipment1VesselName: string;
    public Transshipment1VesselId: string;
    public Transshipment1VesselCode: string;
    public Transshipment1VesselAddedManually: boolean;



    public Transshipment2CarrierName: string;
    public Transshipment2CarrierCode: string;
    public Transshipment2CarrierId: string;
    public Transshipment2CarrierAddedManually: boolean;
    public Transshipment2AirlinePrefix: string;
    public Transshipment2ETD: Date;
    public Transshipment2ETA: Date;
    public Transshipment2ATD: Date;
    public Transshipment2MAWBOBL: string;
    public Transshipment2CarrierNumber: string;
    public Transshipment2VesselName: string;
    public Transshipment2VesselId: string;
    public Transshipment2VesselCode: string;
    public Transshipment2VesselAddedManually: boolean;


    public Transshipment3CarrierName: string;
    public Transshipment3CarrierCode: string;
    public Transshipment3CarrierId: string;
    public Transshipment3CarrierAddedManually: boolean;
    public Transshipment3AirlinePrefix: string;
    public Transshipment3ETD: Date;
    public Transshipment3ETA: Date;
    public Transshipment3ATD: Date;
    public Transshipment3MAWBOBL: string;
    public Transshipment3CarrierNumber: string;
    public Transshipment3VesselName: string;
    public Transshipment3VesselId: string;
    public Transshipment3VesselCode: string;
    public Transshipment3VesselAddedManually: boolean;

    public MoveTypeAddedManually: boolean;
    public MoveTypeId: string;
    public MoveTypeName: string;
    public MoveTypeCode: string;
    public MoveTypeTransportModeId: string;

 
    public ValueOfGoodsCurrencyId: string;
    public ValueOfGoodsCurrencyName: string;
    public ValueOfGoodsCurrencyCode: string;
    public ValueOfGoodsCurrencyAddedManually: boolean;


    ShipmentPickUp: ShipmentPickUpDeliverySL;
    ShipmentDelivery: ShipmentPickUpDeliverySL;

    public get Houses(): Array<HouseSL> {
        if (this.houses == null) {
            this.houses = new Array<HouseSL>();
        }
        return this.houses;
    }
    public set Houses(value: Array<HouseSL>) {
        this.houses = value;
    }
    houses: Array<HouseSL>;

    private shipmentPackages: ShipmentPackagePM[];
    get ShipmentPackages() {
        if (this.shipmentPackages == null) {
            this.shipmentPackages = [];
        }

        return this.shipmentPackages;
    }
    set ShipmentPackages(newValue: ShipmentPackagePM[]) {
        if (this.shipmentPackages != newValue) {
            this.shipmentPackages = newValue;
        }
    }

}

File: Logitude/Logitude.BL/CommonDataModel/DataContracts/ManifestSL.cs
using Logitude.BL.ShipmentsModel.EntityPMs;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Logitude.BL.CommonDataModel.DataContracts
{
    public class ManifestSL
    {

        public string AgentSharedKey { get; set; }
        public string AgentSharedManifestId { get; set; }
        public int SourceAgentTenant { get; set; }
        public int DestinationAgentTenant { get; set; }
        public bool IsDangerous { get; set; }
        
        public double? ValueOfGoods { get; set; }
        public string ShipmentTypeId { get; set; }
        public string ShipmentNumber { get; set; }
        public string MasterNumber { get; set; }
        public DateTime? HAWBDate { get; set; }
        public DateTime? MAWBOBLDate { get; set; }
        public DateTime MasterDate { get; set; }
        public string DirectionId { get; set; }
        public string TransportModeId { get; set; }

        public string IncotermCode { get; set; }
        public string IncotermName{ get; set; }
        public string IncotermId { get; set; }
        public bool IncotermAddedManually { get; set; }

        public string TruckNumber { get; set; }
        public string CarrierCode { get; set; }
        public string CarrierName { get; set; }
        public string CarrierId { get; set; }
        public bool CarrierAddedManually { get; set; }

        public string FreightPrepaidCollectId { get; set; }
        public string OtherPrepaidCollectId { get; set; }
        public string ShipmentLevelCode { get; set; }
        public string HouseNumber { get; set; }
        
        public PartnerSL Consignee { get; set; }
        public string GeneralDescriptionOfGoods { get; set; }
        public PartnerSL Shipper { get; set; }
        public PartnerSL Notify1 { get; set; }
        

        public string GrossWeightUnitCode { get; set; }
        public string ChargeableWeightUnitCode { get; set; }
        public string VolumeUnitCode { get; set; }
        public string DimensionsUnitCode { get; set; }

        public double? ChargeableWeightInKG { get; set; }
        public bool GrossWeightEdited { get; set; }
        public double? GrossWeightInKG { get; set; }

        public double? GrossWeight { get; set; }
        public double? ChargeableWeight { get; set; }
        public double? TEU { get; set; }
        public int? PackagesQuantity { get; set; }

        public double? Volume { get; set; }
        public double? VolumetricWeight { get; set; }
        public int? NumberOfPackages { get; set; }
        public int? NumberOfContainers { get; set; }

        public double? OrderGrossWeight { get; set; }
        public string ShipmentTypeName { get; set; }
   
        public string AgentName { get; set; }

  
        public string AgentReference1 { get; set; }
        public string AgentReference2 { get; set; }

        public string ShipperName { get; set; }
        public string ConsigneeReference1 { get; set; }
        public string ConsigneeReference2 { get; set; }
        public string ShipperReference1 { get; set; }
        public string ShipperReference2 { get; set; }
        public string MainHarmonize { get; set; }


        public string MainCarriageCarrierNumber { get; set; }
        public string Transshipment1CarrierNumber { get; set; }
        public string Transshipment2CarrierNumber { get; set; }
        public string Transshipment3CarrierNumber { get; set; }


        public string MainCarriageAirlinePrefix { get; set; }
        public string MainCarriageVesselName { get; set; }
        public string MainCarriageVesselId { get; set; }
        public string MainCarriageVesselCode { get; set; }
        public bool MainCarriageVesselAddedManually { get; set; }

        public DateTime? MainCarriageETD { get; set; }
        public DateTime? MainCarriageATD { get; set; }
        public DateTime? MainCarriageETA { get; set; }
        public DateTime? MainCarriageATA { get; set; }

        public string MainCarriageMAWBOBL { get; set; }
        public string MainCarriageMAWBOBLDate { get; set; }
        public string InterlineId { get; set; }
        public string InterlineCode { get; set; }
        public string InterlineName { get; set; }
        public bool InterlineAddedManually { get; set; }






        public string Transshipment1CarrierName { get; set; }
        public string Transshipment1CarrierCode { get; set; }
        public string Transshipment1CarrierId { get; set; }
        public bool Transshipment1CarrierAddedManually { get; set; }
        public string Transshipment1AirlinePrefix { get; set; }
        public DateTime? Transshipment1ETD { get; set; }
        public DateTime? Transshipment1ATD { get; set; }

        public DateTime? Transshipment1ETA { get; set; }
        public DateTime? Transshipment1ATA { get; set; }



        public string Transshipment1MAWBOBL { get; set; }

        public string Transshipment1VesselName { get; set; }
        public string Transshipment1VesselId { get; set; }
        public string Transshipment1VesselCode { get; set; }
        public bool Transshipment1VesselAddedManually { get; set; }



        public string Transshipment2CarrierName { get; set; }
        public string Transshipment2CarrierCode { get; set; }
        public string Transshipment2CarrierId { get; set; }
        public bool Transshipment2CarrierAddedManually { get; set; }
        public string Transshipment2AirlinePrefix { get; set; }
        public DateTime? Transshipment2ETD { get; set; }
        public DateTime? Transshipment2ETA { get; set; }
        public DateTime? Transshipment2ATD { get; set; }
        public DateTime? Transshipment2ATA { get; set; }
        public string Transshipment2MAWBOBL { get; set; }
        public string Transshipment2VesselName { get; set; }
        public string Transshipment2VesselId { get; set; }
        public string Transshipment2VesselCode { get; set; }
        public bool Transshipment2VesselAddedManually { get; set; }



        public string Transshipment3CarrierName { get; set; }
        public string Transshipment3CarrierCode { get; set; }
        public string Transshipment3CarrierId { get; set; }
        public bool Transshipment3CarrierAddedManually { get; set; }
        public string Transshipment3AirlinePrefix { get; set; }
        public DateTime? Transshipment3ETD { get; set; }
        public DateTime? Transshipment3ETA { get; set; }
        public DateTime? Transshipment3ATD { get; set; }
        public DateTime? Transshipment3ATA { get; set; }
        public string Transshipment3MAWBOBL { get; set; }
        public string Transshipment3VesselName { get; set; }
        public string Transshipment3VesselId { get; set; }
        public string Transshipment3VesselCode { get; set; }
        public bool Transshipment3VesselAddedManually { get; set; }
        public string TrailerNumber { get; set; }

        

        public PortSL FinalDistenationPort { get; set; }
        public PortSL MainCarriageFromPort { get; set; }
        public PortSL MainCarriageToPort { get; set; }



        public PortSL Transshipment1FromPort { get; set; }
        public PortSL Transshipment1ToPort { get; set; }
        public PortSL Transshipment2FromPort { get; set; }
        public PortSL Transshipment2ToPort { get; set; }
        public PortSL Transshipment3FromPort { get; set; }
        public PortSL Transshipment3ToPort { get; set; }




        public string MoveTypeId { get; set; }
        public string MoveTypeName { get; set; }
        public string MoveTypeCode { get; set; }
        public bool MoveTypeAddedManually { get; set; }
        public string MoveTypeTransportModeId { get; set; }
        

        public string ValueOfGoodsCurrencyId { get; set; }
        public string ValueOfGoodsCurrencyName { get; set; }
        public string ValueOfGoodsCurrencyCode { get; set; }
        public bool   ValueOfGoodsCurrencyAddedManually { get; set; }

        public ShipmentPickUpDeliverySL ShipmentPickUp { get; set; }
        public ShipmentPickUpDeliverySL ShipmentDelivery { get; set; }

        public string LongMaster { get; set; }
        public List<HouseSL> Houses
        {
            get
            {
                if(houses == null)
                {
                    houses = new List<HouseSL>();
                }

                return houses;
            }

            set
            {
                houses = value;
            }
        }
        List<HouseSL> houses;

        private List<ShipmentPackagePM> shipmentPackages;
        public virtual List<ShipmentPackagePM> ShipmentPackages
        {
            get
            {
                if (shipmentPackages == null)
                {
                    shipmentPackages = new List<ShipmentPackagePM>();
                }

                return this.shipmentPackages;
            }

            set
            {
                if (value != null)
                {
                    shipmentPackages = value;
                }
            }
        }



    }

    





}
